src/Ast.ml

```diff
@@ -1441,6 +1444,11 @@ end = struct
     | {pcl_desc= Pcl_fun _}, Non_apply -> true
     | _ -> false
 
+  let memo_parenzed_inner_nested_match =
```

Comment:
Nit: I found the name `memo_` momentarily confusing since this is also used outside `mark_parenzed_inner_nested_match`, so it is more real state than just memoization.

Reply:
renamed